### PR TITLE
Updated overview to include mention of Azure.Identity package.

### LINKED ISSUE #22546: The ProtectKeysWithAzureKeyVault section of this page suggests a series of steps that result in a serious security issue.
The ProtectKeysWithAzureKeyVault section of this page suggests that the reader run the sample code twice.  Once with the ProtectKeysWithAzureKeyVault call commented out to create the initial blob and then a second time with the protect call left in.  The problem with that is that not only is the block file created for the keyring but an initial key is created that is valid for 3 months from the time you ran it.  When you add the ProtectKeysWithAzureKeyVault it actually does NOTHING!.   Your key is now free and clear in blob storage for anyone to read UNTIL that key expires and a new one is generated which is keyvault encrypted.   The instructions should state that the block file should be edited to remove the key or the sample empty file is so small you should just include it in the instructions.  

<?xml version="1.0" encoding="utf-8"?><repository></repository>

The other thing that is wrong is the permissions need for the web app in keyvault, You must have the get permissions as well as wrap and unwrap.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b63ab082-ad30-b00d-ecd1-ca88eee1659e
* Version Independent ID: 69818122-e47a-3fed-ea37-8009f66e2a5c
* Content: [Configure ASP.NET Core Data Protection](https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/configuration/overview?view=aspnetcore-3.1)
* Content Source: [aspnetcore/security/data-protection/configuration/overview.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/data-protection/configuration/overview.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### LINKED ISSUE #24096: Azure.Identity package is required.
Hello,

In order to complete sll the steps in this article, Azure.Identity package is also needed.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b63ab082-ad30-b00d-ecd1-ca88eee1659e
* Version Independent ID: 69818122-e47a-3fed-ea37-8009f66e2a5c
* Content: [Configure ASP.NET Core Data Protection](https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/configuration/overview?view=aspnetcore-5.0)
* Content Source: [aspnetcore/security/data-protection/configuration/overview.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/data-protection/configuration/overview.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by markmcgookin**
Fixes #24096 - By mentioning and linking to the Azure.Identity package, as it is required for keyvault authentication. Also Fixes #22546 where some instructions for using older packages could cause security issues. All reference to old packages has been removed, except mentioning them instructing people to upgrade to the new libraries.

### CODE REVIEW

**Comment by markmcgookin** [comment]
Mentioning #22546 here as well, as the bot doesn't seem to have picked it up from editing the original message.

**Comment by serpent5** [comment]
Hi @markmcgookin. Thanks for this PR. I'll leave it for @Rick-Anderson to review later, but just as an FYI, if you want to link more than one issue, just include `Fixes` for each issue. e.g.:

```
Fixes #24096.
Fixes #22546.
```

-or-

```
Fixes #24096. Fixes #22546.
```

That works when you edit an existing comment too.

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 1]
```diff
@@ -51,25 +51,19 @@ public void ConfigureServices(IServiceCollection services)
         .ProtectKeysWithAzureKeyVault(new Uri("<keyIdentifier>"), new DefaultAzureCredential());
 }
 ```
+In order for your application to communicate and authorise itself with KeyVault, you must also include the [Azure.Identity](https://www.nuget.org/packages/Azure.Identity/) package within your solution.
```
Comment (Rick-Anderson) [inline aspnetcore/security/data-protection/configuration/overview.md:54]:
```suggestion
For an app to communicate and authorize itself with KeyVault,  the [Azure.Identity](https://www.nuget.org/packages/Azure.Identity/) package  must be added.
```

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 2]
```diff
@@ -51,25 +51,19 @@ public void ConfigureServices(IServiceCollection services)
...
-
-Removing `ProtectKeysWithAzureKeyVault` for the first run is advised, as it ensures that the file is created with the proper schema and values in place. 
-
-We recommended upgrading to the [Azure.Extensions.AspNetCore.DataProtection.Blobs](https://www.nuget.org/packages/Azure.Extensions.AspNetCore.DataProtection.Blobs)
-and [Azure.Extensions.AspNetCore.DataProtection.Keys](https://www.nuget.org/packages/Azure.Extensions.AspNetCore.DataProtection.Keys) packages because the API provided automatically creates the blob if it doesn't exist.
+If the app uses the older Azure packages (Microsoft.AspNetCore.DataProtection.AzureStorage and Microsoft.AspNetCore.DataProtection.AzureKeyVault) it is recommended that you ***remove*** these references an upgrade to the [Azure.Extensions.AspNetCore.DataProtection.Blobs](https://www.nuget.org/packages/Azure.Extensions.AspNetCore.DataProtection.Blobs)
+and [Azure.Extensions.AspNetCore.DataProtection.Keys](https://www.nuget.org/packages/Azure.Extensions.AspNetCore.DataProtection.Keys). These packages are where new updates will be provided, and address some key security and stability issues with the older packages.
```
Comment (Rick-Anderson) [inline aspnetcore/security/data-protection/configuration/overview.md:66]:
```suggestion
If the app uses the older Azure packages (Microsoft.AspNetCore.DataProtection.AzureStorage and Microsoft.AspNetCore.DataProtection.AzureKeyVault), we recommend ***removing*** these references and upgrading to the [Azure.Extensions.AspNetCore.DataProtection.Blobs](https://www.nuget.org/packages/Azure.Extensions.AspNetCore.DataProtection.Blobs) and [Azure.Extensions.AspNetCore.DataProtection.Keys] https://www.nuget.org/packages/Azure.Extensions.AspNetCore.DataProtection.Keys). These packages are where new updates are provided, and address some key security and stability issues with the older packages.
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

**Comment by Rick-Anderson** [comment]
@markmcgookin  can you review my suggestions?

image

**Comment by markmcgookin** [comment]
That should be it now @Rick-Anderson

**Pull request merged**